src/components/filters/Filters.component.tsx

```diff
@@ -1,33 +1,99 @@
+import { useState } from 'react';
+
 import searchGlass from '@/assets/filtersImages/searchGlass.svg';
 
 import styles from './Filters.module.css';
 
-export const FiltersComponent = () => (
-    <>
-        <div className={styles.container}>
-            <section className={styles.filterFlex}>
-                <div className={styles.searchContainer}>
-                    <input className={styles.searchBar} type="search" placeholder={'Search...'} />
-                    <button className={styles.searchButton}>
-                        <img src={searchGlass} alt="search" />
-                    </button>
-                </div>
-                <div className={styles.rightFlex}>
-                    <div className={styles.filterButtonsFlex}>
-                        <button className={styles.filterButton}>Electronics</button>
-                        <button className={styles.filterButton}>Shoes</button>
-                        <button className={styles.filterButton}>Clothes</button>
+interface FiltersComponentProps {
+    onCategorySelect: (categories: string[]) => void;
+    onSortSelect: (sortOption: string) => void;
+    onSearch: (searchQuery: string) => void;
+}
+
+export const FiltersComponent: React.FC<FiltersComponentProps> = ({ onCategorySelect, onSortSelect, onSearch }) => {
+    const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+    const [sortOption, setSortOption] = useState<string>('priceSort');
+    const [searchQuery, setSearchQuery] = useState<string>('');
+
+    const handleCategorySelect = (category: string) => {
+        const updatedCategories = selectedCategories.includes(category)
+            ? selectedCategories.filter((cat) => cat !== category)
+            : [...selectedCategories, category];
+        setSelectedCategories(updatedCategories);
+        onCategorySelect(updatedCategories);
+    };
+
+    const handleSortSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const selectedSortOption = event.target.value;
+        setSortOption(selectedSortOption);
+        onSortSelect(selectedSortOption);
+    };
+
+    const handleSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSearchQuery(event.target.value);
+    };
+
+    const handleSearch = () => {
+        onSearch(searchQuery);
+    };
+
+    return (
+        <>
+            <div className={styles.container}>
+                <section className={styles.filterFlex}>
+                    <div className={styles.searchContainer}>
+                        <input
```

Comment:
do not be afraid to create a small components. The search input can be a separate component.